spcconverter.qml

```diff
@@ -571,27 +571,55 @@ MuseScore {
         debugLog("Loaded parameter settings from score.");
     }
 
-    function processAnnotations(texts, staff) {
+    function getVolumeFromDynamic(dynamic) {
+        var volumeMap = {
+            // 255
+            "fff": "FF",
+            // 222
+            "ff": "DE",
+            // 186
+            "f": "BA",
+            // 150
+            "mf": "96",
+            "mp": "96",
+            // 108
+            "p": "6C",
+            // 72
+            "pp": "48",
+            // 36
+            "ppp": "24"
+        };
+        return volumeMap[dynamic];
+    }
```

Comment:
Indexed property access on an object also works on inherited prototype properties, i.e.
```js
volumeMap['constructor']; // returns a function instead of undefined
```

I suggest using a switch block instead, explicitly returning `undefined` in the default case. This is also another reason why the type check on this function's return value should be done with `typeof volume === 'string'` instead of a check for truthiness.

Reply:
The function is only used with the results from `annotations[i].subtypeName()` after the `annotations[i].type` filter. This drastically limits the input into the function already, adding more conditions around that seems a bit excessively defensive IMO. If somehow hidden properties like `__proto__`, `constructor` get in there, the user has done something really, really strange.

That said, I'd be willing to make the change if you think it's necessary. I have some extra time today so I'm going to be going through the code to do a pass on code smells, formatting issues, etc.

Reply:
My reasoning is that it's a public function, meaning that it _could_ be elsewhere in the future. Who knows. Safeguard the future, not the past.